lib/puppet/indirector/file_server.rb

```diff
@@ -29,6 +29,7 @@ def find(request)
     return nil unless path = mount.find(relative_path, request)
     result = model.new(path)
     result.links = request.options[:links] if request.options[:links]
+    result.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
```

Comment:
That doesn't work, because there are some options (source_permissions)
don't exist and it throws an exception. I gave that a try already.

On Fri, Jan 23, 2015 at 2:29 PM, Peter Huene <notifications@github.com>
wrote:

> In lib/puppet/indirector/file_server.rb
> <https://github.com/puppetlabs/puppet/pull/3507#discussion_r23485383>:
>
> > @@ -29,6 +29,7 @@ def find(request)
> >      return nil unless path = mount.find(relative_path, request)
> >      result = model.new(path)
> >      result.links = request.options[:links] if request.options[:links]
> > +    result.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
>
> Also, seems like just passing the options hash along to the constructor of
> the model rather than setting each option explicitly should do the trick,
> although there's one method in Puppet::FileServing::Fileset we don't set
> in the indirector (ignore=). Not sure if we want explicitness where we
> instantiate the model or just pass through the request options.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3507/files#r23485383>.
>